felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1638,19 +1667,60 @@ func (m *bpfEndpointManager) setJumpMapFD(ap attachPoint, fd bpf.MapFD) {
 	})
 }
 
-func (m *bpfEndpointManager) updatePolicyProgram(jumpMapFD bpf.MapFD, rules polprog.Rules) error {
+func (m *bpfEndpointManager) removePolicyDebugInfo(ifaceName string, hook tc.Hook) {
+	if !m.bpfPolicyDebugEnabled {
+		return
+	}
+	filename := bpf.PolicyDebugJSONFileName(ifaceName, string(hook))
+	os.Remove(filename)
+}
+
+func (m *bpfEndpointManager) writePolicyDebugInfo(insns asm.Insns, ifaceName string, tcHook tc.Hook) error {
+	if !m.bpfPolicyDebugEnabled {
+		return nil
+	}
+	if err := os.MkdirAll(bpf.RuntimePolDir, 0600); err != nil {
+		return err
+	}
+
+	polDir := "ingress"
+	if tcHook == tc.HookIngress {
+		polDir = "egress"
+	}
+	var policyDebugInfo = bpf.PolicyDebugInfo{
+		IfaceName:  ifaceName,
+		Hook:       "tc " + string(tcHook),
+		PolicyInfo: insns,
+	}
+
+	filename := bpf.PolicyDebugJSONFileName(ifaceName, polDir)
+	buffer := &bytes.Buffer{}
+	encoder := json.NewEncoder(buffer)
+	encoder.SetEscapeHTML(false)
```

Comment:
Now that the reader is reading the JSON back and formatting it on that side, is this needed?  I'd expect Encode/Decode to be symmetric.